### PR TITLE
MINOR: Upgrade to Gradle 7.3.3

### PR DESCRIPTION
**Pull request by jlprat**
Updates Gradle to its newer version 7.3.3. This version includes the
following relevant features:
- Support for Java 17
- Support for Scala 3

For a further description of the release notes see:
https://docs.gradle.org/7.3.3/release-notes.html

I did the update as per the description in Gradle's release notes:
```
./gradlew wrapper --gradle-version=7.3.3
```

This means `gradlew` script is updated to the newest version.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jlprat** [comment]
Hi @ijuma as you were the last one doing a change like this, do you have time for a review?
Thanks!

**Comment by jlprat** [comment]
Since I opened the PR, a couple of patch versions have been released, I updated this PR to use the latest gradle version `7.3.3`.

**Comment by jlprat** [comment]
As I updated the title of the PR, it seems that the `stage/Build/ARM` build failed. I'll try to close and open again the PR to re-trigger a build if all the others are fine.

**Comment by jlprat** [comment]
Test failures should be unrelated to the change

File: gradlew [hunk 1]
```diff
@@ -17,74 +17,108 @@
...
+  MSYS* | MINGW* )  msys=true    ;; #(
+  NONSTOP* )        nonstop=true ;;
 esac
 
 
 # Loop in case we encounter an error.
 for attempt in 1 2 3; do
   if [ ! -e "$APP_HOME/gradle/wrapper/gradle-wrapper.jar" ]; then
-    if ! curl -s -S --retry 3 -L -o "$APP_HOME/gradle/wrapper/gradle-wrapper.jar" "https://raw.githubusercontent.com/gradle/gradle/v7.2.0/gradle/wrapper/gradle-wrapper.jar"; then
+    if ! curl -s -S --retry 3 -L -o "$APP_HOME/gradle/wrapper/gradle-wrapper.jar" "https://raw.githubusercontent.com/gradle/gradle/v7.3.0/gradle/wrapper/gradle-wrapper.jar"; then
```
Comment (ijuma) [inline gradlew]: This seems to be pointing to 7.3.0 still.

Comment (jlprat) [reply gradlew]: Good catch. I forgot to push this change

Comment (jlprat) [reply gradlew]: it's now fixed @ijuma
thanks for taking a look!

**Comment by jlprat** [comment]
Test failure was https://issues.apache.org/jira/browse/KAFKA-8541 among a couple of other flaky ones.

**Comment by jlprat** [comment]
Ping @ijuma: changes are applied and it should be good to merge

**Review by ijuma — APPROVED** [review]
LGTM, thanks! I also verified that `gradlewAll jar` and `gradlew releaseTarGz` succeed with these changes.

**Comment by jlprat** [comment]
Thanks @ijuma

**Pull request merged**